src/main/java/com/redhat/devtools/lsp4ij/features/semanticTokens/viewProvider/LSPSemanticTokensFileViewProviderHelper.java

```diff
@@ -169,8 +169,8 @@ public void addSemanticToken(@NotNull TextRange textRange,
         if (semanticTokensByOffset != null) {
             LSPSemanticToken semanticToken = new LSPSemanticToken(file, textRange, tokenType, tokenModifiers);
 
-            // Index the token for its text range
-            for (int offset = textRange.getStartOffset(); offset <= textRange.getEndOffset(); offset++) {
+            // Index the token for its text range up to but not including the end offset
+            for (int offset = textRange.getStartOffset(); offset < textRange.getEndOffset(); offset++) {
```

Comment:
The real fix here is `<` instead of `<=`. Basically for a file that uses the semantic tokens file view provider, `file.getElementAt(offset)` should return an element for `[startOffset, endOffset)` where those are the start/end offsets from `element.getTextRange()`. Without this change, it was `[startOffset, endOffset]`, and that was causing an infinite loop in `XDebuggerUtilImpl` which was using `endOffset` as the new offset when searching for the next element.

I'm amazed this wasn't apparent before now!